php/WP_CLI/Process.php

```diff
@@ -46,29 +60,36 @@ private function __construct() {}
 	 * @return ProcessRun
 	 */
 	public function run() {
-		$cwd = $this->cwd;
+		$start_time = microtime( true );
 
-		$descriptors = array(
-			0 => STDIN,
-			1 => array( 'pipe', 'w' ),
-			2 => array( 'pipe', 'w' ),
-		);
-
-		$proc = proc_open( $this->command, $descriptors, $pipes, $cwd, $this->env );
+		$proc = proc_open( $this->command, self::$descriptors, $pipes, $this->cwd, $this->env );
 
 		$stdout = stream_get_contents( $pipes[1] );
 		fclose( $pipes[1] );
 
 		$stderr = stream_get_contents( $pipes[2] );
 		fclose( $pipes[2] );
 
+		$return_code = proc_close( $proc );
+
+		$run_time = microtime( true ) - $start_time;
+
+		if ( getenv( 'WP_CLI_TEST_LOG_RUN_TIMES' ) ) {
```

Comment:
To keep `Process` abstracted from the test suite, could tracking run times instead be a property of the instantiated object?